### PR TITLE
Add flint and steel recipe

### PR DESCRIPTION
**Pull request by AtomicFox556**
#### Summary
Content "Add flint and steel recipe"

#### Purpose of change
Add a way to make flint and steel, an item to start a fire.

#### Describe the solution
Add a recipe for flint and steel.
It turned out that it was impossible to disassemble low/medium/high carbon steel lumps into chunks; this was corrected, though suggestions regarding tools needed for that are welcome.
High carbon steel chunks were also added as a component to all flintlock weapon recipes, as they presumably operate on the same principle as flint and steel and thus would need high carbon steel for the frizzen as well.

#### Describe alternatives you've considered
None.

#### Testing
Launched the game with changes, crafted flint and steel and made sure that it works for starting fires.

#### Additional context
None.

### CODE REVIEW

File: data/json/recipes/tools/tools_hand.json [hunk 1]
```diff
@@ -33,6 +33,26 @@
...
+    "autolearn": true,
+    "using": [ [ "forging_standard", 2 ] ],
+    "proficiencies": [
+      { "proficiency": "prof_metalworking" },
+      { "proficiency": "prof_blacksmithing" },
+      { "proficiency": "prof_toolsmithing" }
+    ],
+    "qualities": [ { "id": "ANVIL", "level": 3 }, { "id": "HAMMER", "level": 3 } ],
+    "tools": [ [ [ "tongs", -1 ] ] ],
+    "components": [ [ [ "hc_steel_chunk", 1 ] ], [ [ "sharp_rock", 1 ] ] ]
```
Comment (kevingranade) [inline data/json/recipes/tools/tools_hand.json:54]: Ehhhhhh..... sharp rock isn't specifically flint, it's just any conchoidal stone, promoting it to flint is quite a stretch.

Comment (AtomicFox556) [reply data/json/recipes/tools/tools_hand.json:54]: My search turned up that in #41711 it was renamed into "flint" with new description specifically mentioning that it makes sparks when struck against steel, and even before that, it was used for making flintlock firearms, so I guess it is meant to be flint specifically. Item ID wasn't changed though.

Before this PR it apparently was a generic rock with an edge usable only for extremely low-quality butchering.

The PR also added an item named "flaking rock" which I think is some rock that won't make sparks but could be used for making stone tools via knapping.

I could rename it back to "sharp rock" and add a new rock named "flint", but then I'm not sure what exactly this "sharp rock" would represent.

**Comment by AtomicFox556** [comment]
Is there anything preventing this from being merged?

**Comment by kevingranade** [comment]
Like I said, treating every flaking stone as flint is a problem, this hasn't been adressed.

**Comment by AtomicFox556** [comment]
Done.

**Comment by AtomicFox556** [comment]
I'll look at the errors later.

**Comment by AtomicFox556** [comment]
Ready now.

**Pull request merged**